src/uuid.c

```diff
@@ -12,7 +12,15 @@
 #define UUID_STRING_LENGTH          36
 #define UUID_STRING_SIZE            (UUID_STRING_LENGTH + 1)
 #define __SUCCESS__                 0
-#define UUID_FORMAT_STRING          "%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x"
+#define HEXA_DIGIT_VAL(c) \
+    ((((c) >= '0') && ((c) <= '9')) \
+        ? ((c) - '0') \
+        : (((c) >= 'a') && ((c) <= 'f')) \
+            ? ((c) - 'a' + 10) \
+            : (((c) >= 'A') && ((c) <= 'F')) ? ((c) - 'A' + 10) : -1)
```

Comment:
```suggestion
#define HEXA_DIGIT_VAL(c)                                             \
    do {                                                              \
        ((((c) >= '0') && ((c) <= '9'))                               \
          ? ((c) - '0')                                               \
          : (((c) >= 'a') && ((c) <= 'f'))                            \
            ? ((c) - 'a' + 10)                                        \
            : (((c) >= 'A') && ((c) <= 'F')) ? ((c) - 'A' + 10) : -1) \
    } while(0)
```

Reply:
The do while zero is useful if there were several functions. In this case we only have a long conditional statement, and it would be ok not to be in the do while zero form.

Reply:
Suggestion to just align the `\`s then for readability!